scripts/install-appstudio-kcp.sh

```diff
@@ -0,0 +1,220 @@
+    echo -e "   -kc, --kcp-context            The name of the kubeconfig context to use."
+    echo -e "   -kk, --kcp-kubeconfig         A valid kcp kubeconfig path."
+    echo -e "   -ck, --cluster-kubeconfig     A valid kubeconfig pointing to a physical openshift cluster."
+    echo -e "   -s, --stable                  Flag to determinate if the kcp cluster is stable or not. In case if this flag is missing by default will install Red Hat App Studio in unstable kcp version.\n"
+    echo -e "Examples:\n"
+    echo -e "   # Deploy Red Hat App Studio in kcp stable version"
+    echo -e "   /bin/bash <E2E_DIR>/scripts/install-appstudio-kcp.sh -kc kcp-stable-root -kk <path-to-kcp-kubeconfig> -ck <path-to-openshift-cluster-kubeconfig> -s\n"
+    echo -e "   # Deploy Red Hat App Studio in kcp unstable version"
+    echo -e "   /bin/bash <E2E_DIR>/scripts/install-appstudio-kcp.sh -kc kcp-unstable-root -kk <path-to-kcp-kubeconfig> -ck <path-to-openshift-cluster-kubeconfig>\n"
+    echo -e "To authenticate against a KCP instance using Red Hat SSO before starting you need to get an offline token from https://console.redhat.com/openshift/token\n"
+    echo -e "   export OFFLINE_TOKEN=<token_goes_here>"
+}
+
+while [[ $# -gt 0 ]]
+do
+    case "$1" in
+        -h|--help)
+            helpUsage
+            exit 0
+            ;;
+        -kc|--kcp-context)
+            export KCP_CONTEXT=$2
+            ;;
+        -kk|--kcp-kubeconfig)
+            export KCP_KUBECONFIG=$2
+            echo $2
+            
+            ;;
+        -ck|--cluster-kubeconfig)
+            export CLUSTER_KUBECONFIG=$2
+            ;;
+        -s|--stable)
+            export IS_STABLE="true"
+            ;;
+        *)
+            ;;
+    esac
+    shift
+done
+
+if [[ -z "$KCP_CONTEXT" ]]; then
+    echo "[ERROR] Not KCP context defined in the script. Please use flag '-kc' or '--kcp-context' to define the kcp context." 
+    helpUsage & exit 1
+fi
+
+if [[ -z "$KCP_KUBECONFIG" ]]; then
+    echo "[ERROR] KCP kubeconfig not defined. Please use flag '-kk' or '--kcp-kubeconfig' to define the kcp kubeconfig." 
+    helpUsage & exit 1
+fi
+
+if [[ -z "$CLUSTER_KUBECONFIG" ]]; then
+    echo "[ERROR] Not cluster kubeconfig defined. Please use flag '-ck' or '--cluster-kubeconfig' to define the kcp physical cluster target kubeconfig." 
+    helpUsage & exit 1
+fi
+
+if [[ -z "$OFFLINE_TOKEN" ]]; then
+    echo "[ERROR] OFFLINE_TOKEN environment is not set. You can obtain one from cloud.redhat.com." 
+    helpUsage & exit 1
+fi
+
+# Export kcp kubeconfig to start working against KCP
+export KUBECONFIG="$KCP_KUBECONFIG"
+
+# Installing oidc-login plugin for kubectl. More information about oidc-login plugin can be found here: https://github.com/int128/kubelogin.
+# oidc-login will be used to authenticate using SSO against KCP
+function installKubectlOIDCLoginPlugin() {
+    echo -e "[INFO] Installing krew for oidc-login plugin installation."
```

Comment:
should there be a code for checking whether oidc plugin is not already installed?

Reply:
The oidc-login plugin install script already do that